src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers/TargetSyntaxMessageLoader.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers
+{
+    public static class TargetSyntaxMessageLoader
+    {
+        /// <summary>
+        /// The suffix that api/message map files are expected to use.
+        /// </summary>
+        private const string TargetSyntaxMessageFileSuffix = ".apitargets";
+
+        private static JsonSerializerOptions GetJsonSerializationOptions()
+        {
+            var jsonSerializerOptions = new JsonSerializerOptions
```

Comment:
Is this available to the analyzer? Dependencies (such as System.Text.Json) are notoriously difficult in roslyn extensions (ie analyzers/source code generators/etc)

Reply:
Yes, this is used to deserialize embedded resources and additional files in the project that define which APIs to create diagnostics for. I thought that I saw this working in VS, but I'm not certain, so I'll double-check.

Reply:
Quick update - I investigated this and as you suspected, the System.Text.Json dependency is problematic depending on how the analyzer is hosted. Basically the same problem as dotnet/roslyn#41785. Things work in UA but it's important that the analyzers work outside of UA, too, so I'll re-work this to not have the System.Text.Json dependency. Ug. :( Good catch, though.

Reply:
Ok, I've reworked this to not use JSON deserialization and confirmed that the analyzers do, in fact, work in VS now.